docs/source/concepts/results.rst

```diff
@@ -1,9 +1,363 @@
+        requires_columns=self.COLUMNS,
+        results_formatter=self.format,
+      )
+
+    def format(self, measure: str, results: pd.DataFrame) -> pd.DataFrame:
+      new_births = results[self.COLUMNS]
+      new_births["birth_date"] = datetime(2024, 12, 30).strftime("%Y-%m-%d T%H:%M.%f")
+      return new_births
+
+As both of these examples are proper ``vivarium``
+:class:`Components <vivarium.component.Component>`, they are added to the simulation
+via the :ref:`model specification <model_specification_concept>` like any other component:
+
+.. code-block:: yaml
+
+  components:
+    <PACKAGE_NAME>:
+      <SUBDIR>:  # as many subdirectories as needed to fully define the path
+        - DeathObserver()
+        - BirthObserver()
+
+Stratifying Observations
+++++++++++++++++++++++++
+
+If you want to stratify the results of an **observation** (that is, group and 
+aggregate by designated categories), you can register a 
+:class:`Stratification <vivarium.framework.results.stratification.Stratification>` 
+with the results system. Stratification registration methods exist on the simulation's 
+:class:`ResultsInterface <vivarium.framework.results.interface.ResultsInterface>` and 
+can be accessed through the :ref:`builder <builder_concept>`:
+
+- :meth:`builder.results.register_stratification <vivarium.framework.results.interface.ResultsInterface.register_stratification>`
+- :meth:`builder.results.register_binned_stratification <vivarium.framework.results.interface.ResultsInterface.register_binned_stratification>`
+
+Here is an example of how you might register a "pregnancy_outcome" stratification 
+as part of the **BirthObserver's** **register_observations** method:
+
+.. testcode::
+
+  from vivarium.framework.engine import Builder
+  from vivarium.framework.results import Observer
+
+  class BirthObserver(Observer):
+
+    ...
+
+    def register_observations(self, builder: Builder) -> None:
+      builder.results.register_stratification(
+        "pregnancy_outcome",
+        ["live_birth", "stillbirth", "pregnancy", "parturition"],
+        requires_columns=["pregnancy_outcome"],
+      )
+      ...  # register observations
+
+    ...
+
+.. note::
+  It is somewhat common to encapsulate all stratification registrations in a single
+  class, though this is not enforced (as the example above demonstrates).
+
+Just because you've *registered* a stratification doesn't mean that the results will
+actually *use* it. In order to use the stratification, you must add it to the 
+:ref:`model specification <model_specification_concept>` configuration block 
+using the "stratification" key. You can provide "default" stratifications which 
+will be used by all observations as well as observation-specific "include" and 
+"exclude" keys to further modify each observation's stratifications.
+
+For example, to use "age_group" and "sex" as default stratifications for *all* 
+observations and then customize the "births" observations to also include 
```

Comment:
The births observation you defined above is a concat observation which is an unstratified observation, so this could be confusing. Can you make this example using the deaths observation instead?

Reply:
Oh yeah, oops - I was trying to be consistent w/ the examples but that's just incorrect